### ISSUE TITLE
Error applying assignment operators to series

### ISSUE THREAD

**Issue by cfobel** [hgrecco/pint-pandas#68]
@hgrecco thanks very much for the recent 0.2 release! I was already making use of the new #47 by installing the dev build, but am happy to see it released 🙂 

I came across some edge cases when using assignment operators and compound assignment operators (e.g., `+=`); particularly with _empty_ slices of a series (see the following code).

```python
>>> import pandas as pd
>>> import pint
>>> import pint_pandas
>>> import pytest
>>> 
>>> 
>>> ureg = pint.get_application_registry()
>>> 
>>> pint_pandas.show_versions()
{'numpy': '1.20.1', 'pandas': '1.1.5', 'pint': '0.17', 'pint_pandas': '0.2'}
>>> durations = pd.Series(1.0, index=range(5))
>>> durations.loc[:] += 1  # OK
>>> durations.loc[[]] += 1  # OK
>>> durations.loc[:] = 1  # OK
>>> durations.loc[[]] = 1  # OK
>>> 
>>> durations = pd.Series(1.0, index=range(5), dtype="pint[meter]")
>>> 
>>> durations[:] += 1 * ureg.meter  # OK
>>> with pytest.raises(IndexError):
...     durations[[]] += 1 * ureg.meter  # Fails: `IndexError: list index out of range`
>>> with pytest.raises(TypeError):
...     durations[:] = 1 * ureg.meter  # Fails: `TypeError: object of type 'int' has no len()`
>>> with pytest.raises(TypeError):
...     durations[[]] = 1 * ureg.meter  # Fails: `TypeError: object of type 'int' has no len()`
```

As shown in the example above, the `=` operator and the `+=` operator both work on a series with `dtype=float`, whether empty or not. However, when using a pint array series, the operator fails under the following conditions:

| Operator | Series | Result |
| -- | -- | -- |
| `=` | Non-empty | ❌ `TypeError` |
| `=` | Empty | ❌ `TypeError` |
| `+=` | Non-empty | ✔️ OK |
| `+=` | Empty | ❌ `IndexError` |

I was able to work around the issue in my case by checking the size of the slice before affected assignment statements, skipping the assignment as needed.

**Comment by hgrecco** [comment]
Actually you should thank @andrewgsavage  !

Please create some tests with this and a PR and we can try to fix it there.

**Comment by andrewgsavage** [comment]
The IndexError can be fixed. I think the += non-empty is tested in arithmetic ops tests from pandas but the += empty can't be so should be added to pandas. I'll open an issue there.

The TypeErrors are due to pandas not handling pint scalaras (the __iter__ / __len__ issue) so aren't as easy to fix. They're tested in TestSetitem.test_setitem_slice and xfailed.

**Comment by cfobel** [comment]
Thanks so much for jumping on this so quickly, @andrewgsavage!

I apologize that I am just getting back to this now. I see #69 which addresses the `+=` compound assignment issue with empty slices; great!

Thanks again for all of your hard work on this issue and on this project in general, @andrewgsavage; I truly appreciate it! 🎉

**Comment by andrewgsavage** [comment]
partially fixed by #69
